client/components/editor.vue

```diff
@@ -465,6 +445,24 @@
           }).fetch();
 
           if (newContents.length) {
+            // Observe new confirmed steps.
+            if (collab.getVersion(this.$editorView.state) > 0) {
+              newContents.filter((x) => {
+                return x.clientId === this.clientId;
+              }).forEach((x) => {
+                const step = Step.fromJSON(schema, x.step);
```

Comment:
You are mapping from JSON twice, here and below. Why don't you process all `newContents` first once, and then iterate over that?

Reply:
Done.